pkgs/by-name/an/anytype/package.nix

```diff
@@ -1,40 +1,118 @@
-    hash = "sha256-inqJvx5K/k97X50E0FYlzJDKqrVjAU6ZKIVdCWHr8NI=";
+  version = "0.44.0";
+  src = fetchFromGitHub {
+    owner = "anyproto";
+    repo = "anytype-ts";
+    rev = "v${version}";
+    sha256 = "sha256-a2ZnTEAFzzTb+lxtQkC6QLG5SP1+gDSjI9dqUNZWfCg=";
   };
-  appimageContents = appimageTools.extractType2 { inherit pname version src; };
-in appimageTools.wrapType2 {
+  description = "P2P note-taking tool";
+
+  locales = fetchFromGitHub {
+    owner = "anyproto";
+    repo = "l10n-anytype-ts";
+    rev = "a5c81ad55383c4e6e9bb7893ecfcb879bac87bea";
+    sha256 = "sha256-evSB0ohHm++tZiazXRMR4vj34IfW3HIkfZ2gwsi/2dk=";
+  };
+in
+buildNpmPackage {
   inherit pname version src;
 
-  nativeBuildInputs = [ makeWrapper ];
+  npmDepsHash = "sha256-DDVsrXgijYYOeCc1gIe2nVb+oL8v4Hqq80d7l5b6MR0=";
+
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    copyDesktopItems
+  ];
+  buildInputs = [ libsecret ];
 
-  extraPkgs = pkgs: [ pkgs.libsecret ];
+  buildPhase = ''
+    runHook preBuild
 
-  extraInstallCommands = ''
-    wrapProgram $out/bin/${pname} \
+    cp -r ${anytype-heart}/lib dist/
+    cp -r ${anytype-heart}/bin/anytypeHelper dist/
+
+    for lang in ${locales}/locales/*; do
+      cp "$lang" "dist/lib/json/lang/$(basename $lang)"
+    done
+
+    npm run build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib/node_modules/anytype
+    cp -r electron.js electron dist node_modules package.json $out/lib/node_modules/anytype/
+
+    for icon in $out/lib/node_modules/anytype/electron/img/icons/*.png; do
+      mkdir -p "$out/share/icons/hicolor/$(basename $icon .png)/apps"
+      ln -s "$icon" "$out/share/icons/hicolor/$(basename $icon .png)/apps/anytype.png"
+    done
+
+    cp LICENSE.md $out/share
+
+    makeWrapper '${lib.getExe electron}' $out/bin/anytype \
+      --set-default ELECTRON_IS_DEV 0 \
       --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}}" \
+      --add-flags $out/lib/node_modules/anytype/ \
       --add-flags ${lib.escapeShellArg commandLineArgs}
-    install -m 444 -D ${appimageContents}/anytype.desktop -t $out/share/applications
-    substituteInPlace $out/share/applications/anytype.desktop \
-      --replace 'Exec=AppRun' 'Exec=${pname}'
-    for size in 16 32 64 128 256 512 1024; do
-      install -m 444 -D ${appimageContents}/usr/share/icons/hicolor/''${size}x''${size}/apps/anytype.png \
-        $out/share/icons/hicolor/''${size}x''${size}/apps/anytype.png
-    done
+
+    runHook postInstall
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Anytype";
+      exec = "anytype";
+      icon = "anytype";
+      desktopName = "Anytype";
+      comment = description;
+      categories = [
+        "Utility"
+        "Office"
+        "Calendar"
+        "ProjectManagement"
+      ];
```

Comment:
Note: It is a valid warning, but I think given the scope of Anytype, that it is acceptable in this case. 
```
Anytype.desktop> /nix/store/m29w56wv1bawarmqd6c1zigwpa7k8l7d-Anytype.desktop/share/applications/Anytype.desktop:
hint: value "Utility;Office;Calendar;ProjectManagement" for key "Categories" in group "Desktop Entry" contains more than one main category;
application might appear more than once in the application menu
```

Reply:
These are taken from the [upstream entry](https://github.com/anyproto/anytype-ts/blob/e65998ee87875a70f3ec4e6c17ed984cc6870964/package.json#L706).